packages/core/database/src/migrations/internal-migrations/5.0.0-05-drop-slug-unique-index.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * In V4 slug fields contained a unique index.
+ * In V5 slug fields should not have a unique index.
+ *
+ * This migration drops existing unique indexes from slug fields so downstream migrations
+ * can work on the data without violating the unique index.
+ */
+import type { Knex } from 'knex';
+
+import type { Migration } from '../common';
+
+const dropIndex = async (knex: Knex, tableName: string, columName: string) => {
+  try {
+    await knex.schema.alterTable(tableName, (table) => {
+      table.dropUnique([columName]);
+    });
+  } catch (error) {
+    // If unique index does not exist, do nothing
+  }
+};
+
+export const dropSlugFieldsIndex: Migration = {
+  name: '5.0.0-05-drop-slug-fields-index',
+  async up(knex, db) {
+    for (const meta of db.metadata.values()) {
+      const hasTable = await knex.schema.hasTable(meta.tableName);
+      if (!hasTable) {
+        continue;
+      }
+
+      for (const attribute of Object.values(meta.attributes)) {
+        if (attribute.type === 'uid' && attribute.columnName) {
+          await dropIndex(knex, meta.tableName, attribute.columnName);
```

Comment:
we don't used a prefix or suffix for the index name ?

Reply:
the table.dropUnique(columnName) removes any unique index tied to that column

Reply:
Does it work with long DB identifiers that are transformed?

Reply:
I assumed (maybe wrongly) the columnName  was the database column name, will check thats the case 🤔

Reply:
I believe `column_name` should be the shortened identifier, and the `convert-identifiers-long-than-max-lenght` should run before this PR migration, so that should be good

Reply:
dropUnique takes a second argument in case the indexName is not the default. Do we actually use the default name 🤔

Reply:
hurm , we do not use the default name, but let me check

Reply:
![image](https://github.com/strapi/strapi/assets/20578351/58d17d4b-7725-488d-a3f6-2ea78bc47849)
Unless this is the default name , I assume we are not using one

but still the migration is running ok 🤔 I will just use the index name just in case

Reply:
looks like knex index unique key matches ours (tableName_columnName_unique)
I created a commit making the index name explicit

Reply:
better safe than sorry 👍